visualization/utils/common.py

```diff
@@ -0,0 +1,388 @@
+from nemo_skills.inference.prompt.few_shot_examples.examples_math import examples_map as examples_math
+from nemo_skills.utils import unroll_files
+
+examples = {
+    **examples_gsm8k,
+    **examples_math,
+}
+
+custom_stats = {}
+general_custom_stats = {}
+deleted_stats = set()
+excluded_rows = set()
+
+
+def get_excluded_row() -> Set:
+    return excluded_rows
+
+
+def get_deleted_stats() -> Set:
+    return deleted_stats
+
+
+def get_custom_stats() -> Dict:
+    return custom_stats
+
+
+def get_general_custom_stats() -> Dict:
+    return general_custom_stats
+
+
+def parse_model_answer(answer: str) -> List[Dict]:
+    code_start, code_end = map(
+        re.escape,
+        current_app.config['prompt_explorer']["visualization_params"]["code_separators"],
+    )
+    output_start, output_end = map(
+        re.escape,
+        current_app.config['prompt_explorer']["visualization_params"]["code_output_separators"],
+    )
+    code_pattern = re.compile(fr'{code_start}(.*?){code_end}', re.DOTALL)
+    code_output_pattern = re.compile(
+        fr'{code_start}((?:(?!{code_end}).)*){code_end}\s*\n\s*{output_start}((?:(?!{output_end}).)*){output_end}',
+        re.DOTALL,
+    )
+    code_matches = list(code_pattern.finditer(answer))
+    code_output_matches = list(code_output_pattern.finditer(answer))
+    parsed_results = []
+    last_index = 0
+    for code_match in code_matches:
+        explanation = answer[last_index : code_match.start()].strip()
+        code_text = code_match.group(1).strip()
+        output_text = None
+        if code_output_matches and code_output_matches[0].start() == code_match.start():
+            output_match = code_output_matches.pop(0)
+            output_text = output_match.group(2).strip()
+        parsed_results.append(
+            {
+                'explanation': explanation,
+                'code': code_text,
+                'output': output_text,
+            }
+        )
+        last_index = code_match.end()
+        if output_text is not None:
+            last_index = output_match.end()
+    if last_index < len(answer):
+        trailing_text = answer[last_index:].strip()
+        if code_start.replace("\\", "") in trailing_text:
+            code_start_index = trailing_text.find(code_start.replace("\\", ""))
+            parsed_results.append(
+                {
+                    'explanation': trailing_text[0:code_start_index].strip(),
+                    'code': trailing_text[code_start_index + len(code_start.replace("\\", "")) :],
+                    'output': "code_block was not finished",
+                    'wrong_code_block': True,
+                }
+            )
+            trailing_text = None
+        if trailing_text:
+            parsed_results.append({'explanation': trailing_text, 'code': None, 'output': None})
+    return parsed_results
+
+
+def get_estimated_height(text) -> float:
+    return max(
+        max(

```

Comment:
please split into multiple expressions, so that black does not autoformat this like that